table/block_based/filter_policy_internal.h

```diff
@@ -31,10 +31,18 @@ class FilterBitsBuilder {
   // added.
   virtual void AddKey(const Slice& key) = 0;
 
+  // Add two entries to the filter, typically a key and, as the alternate,
+  // its prefix. This differs from AddKey(key); AddKey(alt); in that there
+  // is extra state for de-duplicating successive `alt` entries, as well
+  // as successive `key` entries. And there is usually de-duplication between
+  // `key` and `alt` entries, such that it's typically OK for an `alt` without
+  // a corresponding `key` to use AddKey().
```

Comment:
Just for my own learning, if an `alt` directly uses `Addkey`, it will be deduplicated with the last element in the queue, as opposed to the second to last element in the queue, why is that ok for an `alt` without a corresponding `key`?

Reply:
I'm going to change this language to say it's OK to use AddKey on an "alt" as the first and/or last addition to a filter, which is what we need for solving the Seek correctness problem with partitioned filters. As first or last entry, it doesn't interfere with the de-duplication chain of regular keys. :) Thanks!